### PR TITLE
Enhance auth middleware to add userId to request object

### PR DESCRIPTION
**Pull request by alvyjudy**
This PR enhances the auth middleware. It checks and validate the incoming JWT token, then extracts the userId and email from it and add them to the request object. Add it in front of route handlers that require login like the following:
```
const auth = require("path/to/the/file")
router.get("/some/route", auth, (req, res)=>{
   console.log(req.userId);
   console.log(req.email);
}
```

### CODE REVIEW

**Review by kozaktar — APPROVED** [review]
Looks good. I was also thinking that email wouldn't be needed as we can get it from the database at any time

**Comment by alvyjudy** [comment]
That's true. Yea I'll just remove it

**Review by WhoaNellie — APPROVED** [review]

File: server/middleware/auth.js [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+      res.status(401).send("Invalid or expired token");
       return;
-    }else{
-      //console.log(verifiedJwt)
-      next()
+    } else {
+      const decodedToken = jwt(jwtToken);
+      const { user_id, email } = decodedToken;
+      req.userId = user_id;
+      req.email = email; //not sure if email would be needed but I will include here for now
```
Comment (WhoaNellie) [inline server/middleware/auth.js]:
```suggestion
      const { user_id } = decodedToken;
      req.userId = user_id;
```

File: server/middleware/auth.js [hunk 2]
```diff
@@ -1,22 +1,23 @@
 const nJwt = require("njwt");
-const decoder=require("../utils/jwtToken")
+const jwt = require("jwt-decode");
 
-//ensures that user has been authenticated by google
-const auth = async (req, res, next) => {
-  const jwtToken=req.cookies.app_auth_token;
+//This middleware validate the incoming JWT token. Extracts the userId and email
```
Comment (WhoaNellie) [inline server/middleware/auth.js]:
```suggestion
//This middleware validate the incoming JWT token. Extracts the userId.
```

**Pull request merged**